internal/eval/fold_test.go

```diff
@@ -473,6 +473,26 @@ func TestFoldPolicy(t *testing.T) {
 			ast.Permit().When(ast.EntityUID("T", "1").Has("key")),
 			ast.Permit().When(ast.EntityUID("T", "1").Has("key")),
 		},
+		{
+			"opGetTag",
+			ast.Permit().When(ast.Long(42).GetTag(ast.String("key"))),
+			ast.Permit().When(ast.Long(42).GetTag(ast.String("key"))),
+		},
+		{
+			"opHasTag",
+			ast.Permit().When(ast.Long(42).HasTag(ast.String("key"))),
+			ast.Permit().When(ast.Long(42).HasTag(ast.String("key"))),
+		},
+		{
+			"opGetTagEntity",
+			ast.Permit().When(ast.EntityUID("T", "1").GetTag(ast.String("key"))),
+			ast.Permit().When(ast.EntityUID("T", "1").GetTag(ast.String("key"))),
+		},
+		{
+			"opHasTagEntity",
+			ast.Permit().When(ast.EntityUID("T", "1").HasTag(ast.String("key"))),
+			ast.Permit().When(ast.False()),
```

Comment:
@philhassey @patjakdev notice that `hasTag()` nodes are _folded_ when there are no entities in the context. This is because `<non_existing_medallion>.hasTag("someTag")` evaluates to `false`.

Reply:
Isn't this...wrong? Maybe I'm missing something, but we can't be allowed to fold this to false because during folding we don't know which entities actually exist.

The `has` operator works like I expect. Even though it returns false if the entity doesn't exist, it doesn't fold away here...(see `opHasEntity` above).

Reply:
Yes, correct on all accounts! Fixed this up here: https://github.com/cedar-policy/cedar-go/pull/77/commits/4f433e34c428f2de144c83035ba1fd0a313a904c.